sql/updates/post-update.sql

```diff
@@ -171,3 +171,22 @@ $$;
 -- Repair relations that have relacl entries for users that do not
 -- exist in pg_authid
 CALL _timescaledb_functions.repair_relation_acls();
+
+-- Migrate existing CAggs using time_bucket_ng to time_bucket
+DO $$
+DECLARE
+  cagg_name regclass;
+BEGIN
+  FOR cagg_name IN
+    SELECT
+      pg_catalog.format('%I.%I', user_view_schema, user_view_name)::regclass
+    FROM
+      _timescaledb_catalog.continuous_agg
+      JOIN _timescaledb_catalog.continuous_aggs_bucket_function USING (mat_hypertable_id)
+    WHERE
+      bucket_func::text LIKE '%time_bucket_ng%'
+  LOOP
+    CALL _timescaledb_functions.cagg_migrate_to_time_bucket(cagg_name);
+  END LOOP;
+END
+$$;
```

Comment:
We apparently made a decision to do this inside an update script some time in the past. However, I am unsure how expensive the operation is to update the continuous aggregates. This could potentially block an upgrade for a long time if it has to move a lot of data around, or create weird errors if the script is not properly written.

Are we sure this script is not going to block updates for production cases?

Reply:
This function replace the `time_bucket_ng` by `time_bucket` in the cagg query tree... it is very cheap.

Reply:
Not a fan of how this is done. We dont want to overload post-update with things that are version-specific. I think we should do this more surgical.